### PR TITLE
CI: Automatically close broken links issues when new ones are created

### PR DESCRIPTION
**Pull request by inFocus7**
Closes: #5806

## What is the purpose of the change

Improves our developer experience by automatically closing old Broken Links issues when we create new ones during our automated run.

Workflow explanation:
- `only-issue-labels: automated issue,Broken links`: To close only issues with those labels (I added the `Broken links` label to the action to make it more unique)
- `close-issue-message: ...`: To have a message explaining the close, and a link to our open issues - if any - with the labels.
    - The token needs the `issues: write` permission for the message to be written. Unsure if it has those permissions.
- `days-before-stale: -1`: To not mark it as stale, ever
- `days-before-close: 0`: To automatically close the issue. 
    - I could not find examples in the `stale` documentation on if this works, but 'logically' it should. This is the main reason I'd want to see about manually running the workflow to check it works.

## Testing and Verifying

There are no tests for the action, but one way to manually verify if this works is to
1. Reopen https://github.com/osmosis-labs/osmosis/issues/5806 with an added `Broken links` label
2. Run the `Check Broken Links` action with my changes
3. View the run and expect no errors
4. Check that the above issue has been closed on run, with the message linking to our PRs with label(s).

Note: It's possible that no new issue gets created if there are no broken links. We can confirm it works as expected by ensuring there are no failures during the CI run. We know the action works since it has created issues before, so the functionality we care about is the new step `Close Existing Broken Links Issue(s)` not breaking the CI proccess.

## Documentation and Release Note

  - [ ] Does this pull request introduce a new feature or user-facing behavior changes?
  - [x] Changelog entry added to `Unreleased` section of `CHANGELOG.md`?

Where is the change documented? 
  - [ ] Specification (`x/{module}/README.md`)
  - [ ] Osmosis documentation site
  - [ ] Code comments?
  - [ ] N/A

### CODE REVIEW

File: .github/workflows/check-broken-links.yml
Comment (mattverse) [inline .github/workflows/check-broken-links.yml:20]: Is this link suppose to take us somewhere?

https://github.com/osmosis-labs/osmosis/issues?q=is%3Aissue+is%3Aopen+label%3A%22automated+issue%22+label%3A%22Broken+links%2

File: .github/workflows/check-broken-links.yml
Comment (mattverse) [inline .github/workflows/check-broken-links.yml:21]: what does -1 stand for? I'm guessing nil?

**Review by pysel — COMMENTED** [review]
Is it maybe possible to somehow check on a fork? or maybe it would be nice to make it possible to run this action manually and check that it indeed works after it gets merged

File: .github/workflows/check-broken-links.yml
Comment (inFocus7) [reply .github/workflows/check-broken-links.yml:20]: Ah I think you're missing the last `2` in the link. It's a link to our open issues filtered with both the label `automated issue` and `Broken links`. So if there's a new issue generated, people can see it there.

File: .github/workflows/check-broken-links.yml
Comment (inFocus7) [reply .github/workflows/check-broken-links.yml:21]: Yeah, it's so that the issue is never marked as `stale`
> If set to a negative number like -1, the issues or the pull requests will never be closed automatically.

My idea is to not mark the issue as stale `-1`, _but_ do close it (`0`) when this action is ran, since a new issue should be created with the latest broken links.

**Comment by inFocus7** [comment]
@pysel The action has a `workflow_dispatch`, so you should be able to manually run it on the action [here](https://github.com/osmosis-labs/osmosis/actions/workflows/check-broken-links.yml) - if you have the permissions to. Although it would have to be post-merge. I tried seeing if I can run it from my fork but the action doesn't appear.

**Review by mattverse — APPROVED** [review]
I'm in favor of merging this and giving it a try

**Comment by inFocus7** [comment]
@mattverse awesome! If we manually kick it off, we'll want to make sure the latest broken links issue is open and has "Broken links" label. Once we run it, if it works, it'll close that issue and create a new issue.

If y'all can manually run it from the actions tab, i think its possible to run the action from a specific branch. That way we don't merge this until verifying the action from this branch.

**Pull request merged**
